### ISSUE TITLE
Newly created files don't belong to ts-project

### ISSUE THREAD

**Issue by jrieken** [microsoft/TypeScript#35794]
*TS Issue template filled out by @mjbvz from original report*

**TypeScript Version**: 3.7.3
 
**Repo steps:**
1. In the VS Code code base
1. Open `src/vs/workbench/contrib/codeActions/common/codeActions.contribution.ts`
1. Wait for JS/TS to finish initializing
1. Select the lines:

```ts
Registry.as<IConfigurationRegistry>(Extensions.Configuration)
	.registerConfiguration(editorConfiguration);
```

4. Run the `move to new file` refactoring

**Bug**
The newly created file is not part of any project

**Comment by mjbvz** [comment]
Here's a [tsserver.log](https://github.com/microsoft/TypeScript/files/3986343/tsserver.log) for the above repo. The important bit is that `newFile.ts` is stuck in a new inferred project instead of the main project:

```
Info 2398 [17:57:12.924] Open files: 
Info 2398 [17:57:12.924] 	FileName: /Users/matb/projects/vscode/src/vs/workbench/contrib/codeActions/common/codeActions.contribution.ts ProjectRootPath: /Users/matb/projects/vscode
Info 2398 [17:57:12.924] 		Projects: /Users/matb/projects/vscode/src/tsconfig.json
Info 2398 [17:57:12.924] 	FileName: /Users/matb/projects/vscode/src/vs/workbench/contrib/codeActions/common/newFile.ts ProjectRootPath: /Users/matb/projects/vscode
Info 2398 [17:57:12.925] 		Projects: /dev/null/inferredProject1*
```

**Comment by mjbvz** [comment]
Just tested this in typescript: 3.8.0-dev.20191219 and it seems to work properly

@jrieken Can you also please try upgrading to use `typescript@next` by installing [this extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next). Try using that version of TS for a few days while working on VS Code and let us know if you hit the problem again

**Comment by jrieken** [comment]
This also happens with 3.8.0-dev.20201009

**Comment by sheetalkamat** [comment]
I don't seem to repro this... 

[tsserver.log](https://github.com/microsoft/TypeScript/files/4047201/tsserver.log)

Can you get the latest tsserver log, may be it will reveal something.
```
Info 2348 [10:35:38.323] Open files: 
Info 2348 [10:35:38.323] 	FileName: c:/github/vscode/src/vs/workbench/contrib/codeActions/common/codeActions.contribution.ts ProjectRootPath: c:/github/vscode
Info 2348 [10:35:38.323] 		Projects: c:/github/vscode/src/tsconfig.json
Info 2348 [10:35:38.323] 	FileName: c:/github/vscode/src/vs/workbench/contrib/codeActions/common/newFile.ts ProjectRootPath: c:/github/vscode
Info 2348 [10:35:38.323] 		Projects: c:/github/vscode/src/tsconfig.json
```

**Comment by jrieken** [comment]
[tsserver.log](https://github.com/microsoft/TypeScript/files/4052572/tsserver.log)

There you go - create `callHierarchtFoo.ts` and copied this import `import { IModelService } from 'vs/editor/common/services/modelService';` which yields in a compile error. Using ts 3.8.0-dev-20200111

**Comment by sheetalkamat** [comment]
@jrieken Thank you for the log.. This seems to arise only if the file creation watcher is invoked after file is opened (because of some perf optimized code we have on how we handle new file that is open created/deleted).

**Comment by realh** [comment]
> This seems to arise only if the file creation watcher is invoked after file is opened (because of some perf optimized code we have on how we handle new file that is open created/deleted).

So it's some sort of race condition? In #88787 I mentioned my project was on a sshfs mount (via wifi). Could the lag from the filing system be exacerbating it?

**Comment by BetterCallSky** [comment]
@sheetalkamat 
Is it possible to include this fix to 3.7? 🙏
As described here https://github.com/microsoft/vscode/issues/88660, this issue is very frustrating for me. Sometimes, I must restart TS from the command pallet 20 times per day 😕

**Comment by orta** [comment]
Independent of the release, you can use [this extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next) to use an up-to-date copy of the TS tooling for VS code. ( Once the PR is merged )

**Comment by BetterCallSky** [comment]
Does `3.8.0-dev.20200125` include this commit or I have to wait for `0126`? I was able to repeat this issue in `0125`.

**Comment by BetterCallSky** [comment]
@orta @sheetalkamat 
The issue is not fixed. I am able to reproduce it in `3.8.0-dev.20200128`.

**Comment by sheetalkamat** [comment]
@BetterCallSky The original repro is fixed. I verified it again just to make sure. You would need to create new issue with repro to be able to investigate this.

**Comment by BetterCallSky** [comment]
@sheetalkamat 
I created here https://github.com/microsoft/vscode/issues/88660, but it's closed.

https://github.com/microsoft/TypeScript/issues/35794#issuecomment-567755209
My tsserver.log is very similar. When I create a new file, VSCode/TS creates a new project for it.

**Comment by sheetalkamat** [comment]
You would want to create new issue in typescript repo with repro and tsserver log.
